### PR TITLE
Identity Library Updates

### PR DESCRIPTION
**Pull request by g2vinay**
Renames AzureCredential to DefaultAzureCredential
Renames ChainedCredential to ChainedTokenCredential

Removes MsiEndpoint and MsiSecret setters from ManagedIdentityCredential.
Removes IdentityId and ObjectId from ManagedIdentityCredential, VirtualMachineMSICredential
Plumbs ClientId to AppServiceMSICredential into authrequest (query param)
Makes AppServiceMSICredential and VirtualMachineMSICredential package private and moves them to credential package.

### CODE REVIEW

File: identity/client/src/main/java/com/azure/identity/IdentityClient.java [hunk 1]
```diff
@@ -149,19 +149,26 @@ private static AuthenticationContext createAuthenticationContext(ExecutorService
...
         String resource = ScopeUtil.scopesToResource(scopes);
         HttpURLConnection connection = null;
         try {
-            String urlString = String.format("%s?resource=%s&api-version=2017-09-01", msiEndpoint, resource);
-            URL url = new URL(urlString);
+            StringBuilder urlStringBuilder = new StringBuilder(String.format("%s?resource=%s&api-version=2017-09-01", msiEndpoint, resource));
+            if (clientId != null) {
+                urlStringBuilder.append("&");
+                urlStringBuilder.append("client_id");
+                urlStringBuilder.append("=");
```
Comment (JonathanGiles) [inline identity/client/src/main/java/com/azure/identity/IdentityClient.java]: These three lines can be made into one line - you're concatenating three string constants.

File: identity/client/src/main/java/com/azure/identity/IdentityClient.java [hunk 2]
```diff
@@ -149,19 +149,26 @@ private static AuthenticationContext createAuthenticationContext(ExecutorService
...
      * @return a Publisher that emits an AccessToken
      */
-    public Mono<AccessToken> authenticateToManagedIdentityEnpoint(String msiEndpoint, String msiSecret, String[] scopes) {
+    public Mono<AccessToken> authenticateToManagedIdentityEnpoint(String msiEndpoint, String msiSecret, String clientId, String[] scopes) {
         String resource = ScopeUtil.scopesToResource(scopes);
         HttpURLConnection connection = null;
         try {
-            String urlString = String.format("%s?resource=%s&api-version=2017-09-01", msiEndpoint, resource);
-            URL url = new URL(urlString);
+            StringBuilder urlStringBuilder = new StringBuilder(String.format("%s?resource=%s&api-version=2017-09-01", msiEndpoint, resource));
```
Comment (JonathanGiles) [inline identity/client/src/main/java/com/azure/identity/IdentityClient.java]: It's weird that you're using string.format within a string builder. You should do only one (stringbuilder I would say).

File: identity/client/src/main/java/com/azure/identity/IdentityClient.java [hunk 3]
```diff
@@ -204,21 +209,11 @@ private static AuthenticationContext createAuthenticationContext(ExecutorService
...
-                payload.append("&");
-                payload.append("object_id");
-                payload.append("=");
-                payload.append(URLEncoder.encode(objectId, "UTF-8"));
-            } else if (clientId != null) {
+            if (clientId != null) {
                 payload.append("&");
                 payload.append("client_id");
                 payload.append("=");
                 payload.append(URLEncoder.encode(clientId, "UTF-8"));
```
Comment (JonathanGiles) [inline identity/client/src/main/java/com/azure/identity/IdentityClient.java]: Same issue here in code that isn't changed - excessive string builder operations that could be made simpler

File: identity/client/src/main/java/com/azure/identity/IdentityClient.java (hunk 2 above)
Comment (schaabs) [inline identity/client/src/main/java/com/azure/identity/IdentityClient.java]: >String.format("%s?resource=%s&api-version=2017-09-01", msiEndpoint, resource) [](start = 63, length = 77)

Not sure if URL handles this but it's most likely that the resource needs to be url encoded as a data string.

Comment (g2vinay) [reply identity/client/src/main/java/com/azure/identity/IdentityClient.java]: added encoding

File: identity/client/src/main/java/com/azure/identity/IdentityClient.java (hunk 3 above)
Comment (g2vinay) [reply identity/client/src/main/java/com/azure/identity/IdentityClient.java]: made simpler

File: identity/client/src/main/java/com/azure/identity/IdentityClient.java (hunk 2 above)
Comment (g2vinay) [reply identity/client/src/main/java/com/azure/identity/IdentityClient.java]: Updated.
StringBuilder not needed here anymore

File: identity/client/src/main/java/com/azure/identity/IdentityClient.java (hunk 1 above)
Comment (g2vinay) [reply identity/client/src/main/java/com/azure/identity/IdentityClient.java]: made simpler

**Pull request merged**